### PR TITLE
Task 4

### PR DESCRIPTION
**Pull request by maximsan**
lambda for CRUD operations  - [lambda](https://0nw7hvxew9.execute-api.eu-west-1.amazonaws.com/dev/products)
FE pull-request - [link](https://github.com/maximsan/nodejs-aws-fe/pull/2)
Cloudfront - [link](dg2e8nno6k0ty.cloudfront.net)
POST/products lambda functions returns error 400 status code if product data is invalid
All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
All lambdas do console.log for each incoming requests and their arguments

### CODE REVIEW

**Comment by asamurai** [comment]
Снимок экрана 2020-11-09 в 12 48 05
product list link returns internal server error
please fix it

also provide please link on your CloudFront

File: product-service/src/utils/errorHandler.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import {BadRequest} from "./BadRequest";
+import {createResponse} from "./createResponse";
+
+export const errorHandler = (error) => {
+    console.log(`error in errorHandler ${JSON.stringify(error)}`);
+
+    if (error instanceof NotFoundError) {
+        return createResponse(404, error.message);
+    }
+    if (error instanceof BadRequest) {
```
Comment (IgorByl) [inline product-service/src/utils/errorHandler.js:11]: checking of the instances' classes goes against solid

Comment (maximsan) [reply product-service/src/utils/errorHandler.js:11]: Thanks for review. SOLID includes 5 statements and your assertion looks a little surface. Which solid statement does it ruin? The same approach we use when checking objects against interfaces. Maybe you have something to propose ?

**Comment by maximsan** [comment]
> Снимок экрана 2020-11-09 в 12 48 05
> 
> product list link returns internal server error please fix it
> also provide please link on your CloudFront

It was database related issue. I changed pass. It's working. Cloudfront link is in FE pr, but I've added it as well.

File: product-service/src/utils/errorHandler.js (hunk 1 above)
Comment (IgorByl) [reply product-service/src/utils/errorHandler.js:11]: Hi, the Liskov Substitution Principle restricts class instance type checking.

**Pull request merged**